sceptre/hooks/__init__.py

```diff
@@ -37,6 +43,15 @@ def run(self):
         """
         pass  # pragma: no cover
 
+    def clone(self, stack: "Stack") -> "Hook":
+        """
+        Produces a "fresh" copy of the Hook, with the specified stack.
+
+        :param stack: The stack to set on the cloned resolver
+        """
+        clone = type(self)(self.argument, stack)
+        return clone
+
 
```

Comment:
This makes hooks operate more like Resolvers so we have more consistent behavior all around.